### ISSUE TITLE
summaryLength doesn't count headers, lists, or code blocks as words

### ISSUE THREAD

**Issue by akbyrd** [gohugoio/hugo#11863]
### What version of Hugo are you using (`hugo version`)?
0.122.0

### Does this issue reproduce with the latest release?
Yes

Given a config with `summaryLength = 2` and a post with the following content
```
asd

## Header
- TODO: 1
- TODO: 2
- TODO: 3

## Header
- TODO: 1
- TODO: 2
- TODO: 3

## Header
- TODO: 1
- TODO: 2
- TODO: 3
```

The summary will be much longer than 2 words
> asd Header TODO: 1 TODO: 2 TODO: 3 Header TODO: 1 TODO: 2 TODO: 3 Header TODO: 1 TODO: 2 TODO: 3

Inserting a second `asd` somewhere will truncate the summary sooner.
```
asd

## Header
asd
- TODO: 1
- TODO: 2
- TODO: 3

## Header
- TODO: 1
- TODO: 2
- TODO: 3

## Header
- TODO: 1
- TODO: 2
- TODO: 3
```
> asd Header asd

From this, I gather that headers and lists are not counted as words and can lead to arbitrarily long summaries. I think they should both count toward the word limit.

Interestingly, lists _sometimes_ count towards the word limit.
```
asd

## Header
- TODO: 1
asd
- TODO: 2
- TODO: 3
```
> asd Header TODO: 1 

In this case `TODO: 1` has been counted as a word and the second `asd` was not included in the summary.

This also affects code blocks. Entire code blocks of arbitrary size are included in summaries without affecting the world limit.

**Comment by jmooring** [comment]
TLDR: Use a [manual summary](https://gohugo.io/content-management/summaries/#manual-summary-splitting) in this case.

### Automatic summary

When rendering an [automatic summary](https://gohugo.io/content-management/summaries/#automatic-summary-splitting), Hugo uses the `summaryLength` as a minimum and attempts to truncate at a sensible point in the text (e.g., at the end of a sentence).

#### Example 1

`summaryLength =2`

content

```
The quick brown fox jumps over the lazy dog. Consequat dolor cupidatat adipisicing consequat ut id proident amet id excepteur deserunt. Minim aute reprehenderit cupidatat ea magna mollit amet sunt officia id consequat id deserunt.
```

summary (data type = template.HTML)

```
The quick brown fox jumps over the lazy dog.
```

#### Example 2

`summaryLength = 9`

content

```
The quick brown fox jumps over the lazy dog. Consequat dolor cupidatat adipisicing consequat ut id proident amet id excepteur deserunt. Minim aute reprehenderit cupidatat ea magna mollit amet sunt officia id consequat id deserunt.
```

summary (data type = template.HTML)

```html
The quick brown fox jumps over the lazy dog. Consequat dolor cupidatat adipisicing consequat ut id proident amet id excepteur deserunt.
```

### Manual summary

To truncate at a specific point in the content, use a [manual summary](https://gohugo.io/content-management/summaries/#manual-summary-splitting) (`summaryLength` is irrelevant).

content

```
The quick <!--more--> brown fox jumps over the lazy dog. Consequat dolor cupidatat adipisicing consequat ut id proident amet id excepteur deserunt. Minim aute reprehenderit cupidatat ea magna mollit amet sunt officia id consequat id deserunt.
```

summary (data type = template.HTML)

```html
<p>The quick</p>
```

### Summary in front matter

Or you can specify the summary in front matter (`summaryLength` and manual break are irrelevant).

```
+++
title = 'Something'
date = 2024-01-06T08:46:33-08:00
draft = false
summary = 'The quick brown fox'
+++
```

summary (data type = template.HTML)

```html
<p>The quick brown fox</p>
```

### Other

This [comment](https://github.com/gohugoio/hugo/issues/8910#issuecomment-903158600) describes some of the other differences between the 3 summary types.

### Opinion

In my view there are 3 different related things:

1. A _teaser_ (the first N sentences,)
2. A _description_ (one sentence, typically used in meta data)
3. A _summary_ (very different than a teaser)

For this reason I usually define both `description` and `summary` in front matter. If I don't define the `summary` in front matter, I always use a manual summary to precisely control the result.

I am closing this because the existing behavior is expected.

**Comment by akbyrd** [comment]
Manual and front matter summaries are a reasonable workaround that I've already switched to using. However it seems quite reasonable to say that `summaryLength` does not work properly or consistently. Being able to set a word limit then have it ignored means that option is non-functional.

It might be reasonable to treat this as dupe of https://github.com/gohugoio/hugo/issues/1503, since that may or may not end up addressing this.

**Comment by jmooring** [comment]
We will handle this in documentation, see <https://github.com/gohugoio/hugoDocs/issues/2388>.

The `summaryLength`, as described above, is not a word limit. With that in mind, and the subtleties of finding a sensible breaking point, the existing behavior is working properly and consistently.
